### DISCUSSION TITLE
Performance issues

### DISCUSSION THREAD

**Discussion by adam-qiang** [laravel/framework#52621] [Q&A]
### Laravel Version

11.20.0

### PHP Version

8.3.10

### Database Driver & Version

MySQL5.7

### Description

There are performance issues when using Schema:: hasTable(), which can generate slow SQL and cause a surge in database load (see performance report in the following figure)

![image](https://github.com/user-attachments/assets/5bdcbea0-3d87-4048-af4b-777572a6c6f5)

### Steps To Reproduce

Schema:: hasTable()

**Comment by alecpl** [comment]
`information_schema.tables` is the most standard way of doing that, but it is a view on a multiple tables and it can be slow indeed. If we really wanted to change that using some system tables directly might be one solution. There's also [table_exists](https://dev.mysql.com/doc/refman/5.7/en/sys-table-exists.html) added in MySQL 5.7, but I have no idea about its performance.

That being said, I use `Schema` features only in migration where the performance is not that big of a deal.

**Comment by adam-qiang** [comment]
> `information_schema.tables` is the most standard way of doing that, but it is a view on a multiple tables and it can be slow indeed. If we really wanted to change that using some system tables directly might be one solution. There's also [table_exists](https://dev.mysql.com/doc/refman/5.7/en/sys-table-exists.html) added in MySQL 5.7, but I have no idea about its performance.
> 
> That being said, I use `Schema` features only in migration where the performance is not that big of a deal.

If it's just a simple way to determine whether a table exists, and whether there is a more lightweight function? (I've written a function myself as a substitute for it)
